exec-service-wasmer/src/wasmer_breakpoint.rs

```diff
@@ -0,0 +1,168 @@
+use std::mem;
+use std::sync::Mutex;
+
+use loupe::{MemoryUsage, MemoryUsageTracker};
+use wasmer::wasmparser::{Operator, Type as WpType, TypeOrFuncType as WpTypeOrFuncType};
+use wasmer::{
+    ExportIndex, FunctionMiddleware, GlobalInit, GlobalType, Instance, LocalFunctionIndex,
+    MiddlewareError, MiddlewareReaderState, ModuleMiddleware, Mutability, Type,
+};
+use wasmer_types::{GlobalIndex, ModuleInfo};
+
+const BREAKPOINT_VALUE: &str = "breakpoint_value";
+
+pub(crate) const BREAKPOINT_VALUE_NO_BREAKPOINT: u64 = 0;
+#[allow(dead_code)]
+pub(crate) const BREAKPOINT_VALUE_EXECUTION_FAILED: u64 = 1;
+pub(crate) const BREAKPOINT_VALUE_OUT_OF_GAS: u64 = 4;
+#[allow(dead_code)]
+pub(crate) const BREAKPOINT_VALUE_MEMORY_LIMIT: u64 = 5;
+
+#[derive(Clone, Debug, MemoryUsage)]
+struct BreakpointGlobalIndex(GlobalIndex);
+
+impl BreakpointGlobalIndex {
+    fn breakpoint_value(&self) -> GlobalIndex {
+        self.0
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct Breakpoint {
```

Comment:
`Breakpoints`, plural.

Reading `struct Breakpoint` suggest that this struct somehow stands for a single breakpoint.

Reply:
done